src/platforms/atomic-kms/server/gbm_display_allocator.cpp

```diff
@@ -160,9 +171,16 @@ auto create_gbm_surface(gbm_device* gbm, geom::Size size, mg::DRMFormat format,
 class GBMSurfaceImpl : public mga::GBMDisplayAllocator::GBMSurface
 {
 public:
-    GBMSurfaceImpl(mir::Fd drm_fd, gbm_device* gbm, geom::Size size, mg::DRMFormat const format, std::span<uint64_t> modifiers)
-        : drm_fd{std::move(drm_fd)},
-          surface{create_gbm_surface(gbm, size, format, modifiers)}
+    GBMSurfaceImpl(
+        mir::Fd drm_fd,
+        gbm_device* gbm,
+        geom::Size size,
+        mg::DRMFormat const format,
+        std::span<uint64_t> modifiers,
+        std::shared_ptr<mga::GbmQuirks> runtime_quirks) :
+        drm_fd{std::move(drm_fd)},
+        runtime_quirks{runtime_quirks},
+        surface{create_gbm_surface(gbm, size, format, modifiers, runtime_quirks)}
     {
     }
```

Comment:
Naming and excess copying of `shared_ptr<mga::GbmQuirks>`...
```suggestion
    GBMSurfaceImpl(
        mir::Fd drm_fd,
        gbm_device* gbm,
        geom::Size size,
        mg::DRMFormat const format,
        std::span<uint64_t> modifiers,
        std::shared_ptr<mga::GbmQuirks> const& gbm_quirks) :
        drm_fd{std::move(drm_fd)},
        gbm_quirks{gbm_quirks},
        surface{create_gbm_surface(gbm, size, format, modifiers, *gbm_quirks)}
    {
    }
```